libp2p-pki/src/lib.rs

```diff
@@ -0,0 +1,182 @@
+extern crate asn1;
+
+use std::fmt;
+
+pub enum Modulo {
```

Comment:
I don't know much about RSA and stuff, but is there a way to make this more generic over the key size?

Reply:
https://docs.rs/generic-array/0.9.0/generic_array/ but it's a little weird to use

Reply:
PR to typenum so we can bound it by powers of two: https://github.com/paholg/typenum/pull/92

Reply:
typenum PR has been merged and the crate released since. Do we want to get back onto this?